hypothesis-python/src/hypothesis/extra/django/_fields.py

```diff
@@ -299,7 +344,7 @@ def from_field(field: F) -> st.SearchStrategy[Union[F, None]]:
     instance attributes such as string length and validators.
     """
     check_type((dm.Field, df.Field), field, "field")
-    if getattr(field, "choices", False):
+    if not isinstance(field, df.ModelChoiceField) and getattr(field, "choices", False):
```

Comment:
The `isinstance` check needs to happen first because `getattr(field, "choices")` causes database access.

Reply:
Let's make that a code-comment, so that future maintainers don't break the invariant by accident.

(I believe some of the new tests would also fail, which is fantastic but provides slower and murkier feedback)

Reply:
Added in 33abde525